lib/foreman/observable.rb

```diff
@@ -4,10 +4,10 @@ module Observable
 
     def trigger_hook(hook_name, namespace: DEFAULT_NAMESPACE, payload: nil, &blk)
       event_name = event_name_for(hook_name, namespace: namespace)
-      event_context = { context: ::Logging.mdc.context.symbolize_keys }
-      event_payload = event_payload_for(payload, blk).to_h
-      payload_with_context = event_payload.merge(event_context)
-      ActiveSupport::Notifications.instrument(event_name, payload_with_context)
+      event_payload = { context: ::Logging.mdc.context.symbolize_keys }.with_indifferent_access
+      yielded_payload = event_payload_for(payload, blk)
+      event_payload.merge!(yielded_payload) if yielded_payload
+      ActiveSupport::Notifications.instrument(event_name, event_payload)
     end
```

Comment:
I had to refactor this because we want a clean way to create a hash with indifferent access in all cases (e.g. when payload or block yields to regular hash).